### DISCUSSION TITLE
Different platforms and determinism

### DISCUSSION THREAD

**Discussion by Mapiarz** [google/or-tools#3855] [CP-SAT questions] [answered]
Hi!

I have a test set where basically I have a set of known inputs and the expected results. We run CP-SAT with just 1 worker and the results are deterministic. Or so we thought. Some tests fail for my one of my colleagues. He's the only one running Mac OS with the new M chip. Could that be the culprit of why some of the tests fail for them? So far we've tested this on multiple x64 machines running Ubuntu and had no problems. Please advise.

Thanks!

**Comment by lperron** [comment, marked as answer]
Floats, timing, hashing do not behave the same on different platforms.

You can also check that the models are the same. We display a checksum in the log.
